src/main/io/gps.c

```diff
@@ -543,10 +544,14 @@ static void ubloxSetMessageRate(uint8_t messageClass, uint8_t messageID, uint8_t
     ubloxSendConfigMessage(&tx_buffer, MSG_CFG_MSG, sizeof(ubxCfgMsg_t));
 }
 
-static void ubloxSetNavRate(uint16_t measRate, uint16_t navRate, uint16_t timeRef)
-{
+static void ubloxSetNavRate(uint8_t measRateHz, uint16_t navRate, uint16_t timeRef) {
     ubxMessage_t tx_buffer;
-    tx_buffer.payload.cfg_rate.measRate = measRate;
+
+    uint8_t measRateMilliseconds = 1000/measRateHz;
```

Comment:
```suggestion
    uint8_t measRateMilliseconds = 1000 / measRateHz;
```

Also, `measRateHz < 4 will overflow